ibm/service/power/resource_ibm_pi_volume_group_test.go

```diff
@@ -0,0 +1,141 @@
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckIBMPIVolumeGroupConfig(name),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckIBMPIVolumeGroupExists("ibm_pi_volume_group.power_volume_group"),
+					resource.TestCheckResourceAttr(
+						"ibm_pi_volume_group.power_volume_group", "pi_volume_group_name", name),
+				),
+			},
+		},
+	})
+}
+
+func TestAccIBMPIVolumeGroupUpdate(t *testing.T) {
+	name := fmt.Sprintf("tf-pi-volume-group-%d", acctest.RandIntRange(10, 100))
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { acc.TestAccPreCheck(t) },
+		Providers:    acc.TestAccProviders,
+		CheckDestroy: testAccCheckIBMPIVolumeGroupDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckIBMPIVolumeGroupConfig(name),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckIBMPIVolumeGroupExists("ibm_pi_volume_group.power_volume_group"),
+					resource.TestCheckResourceAttr(
+						"ibm_pi_volume_group.power_volume_group", "pi_volume_group_name", name),
+				),
+			},
+			{
+				Config: testAccCheckIBMPIVolumeGroupUpdateConfig(name),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckIBMPIVolumeGroupExists("ibm_pi_volume_group.power_volume_group"),
+					resource.TestCheckResourceAttr(
+						"ibm_pi_volume_group.power_volume_group", "pi_volume_group_name", name),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckIBMPIVolumeGroupDestroy(s *terraform.State) error {
+
+	sess, err := acc.TestAccProvider.Meta().(conns.ClientSession).IBMPISession()
+	if err != nil {
+		return err
+	}
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "ibm_pi_volume_group" {
+			continue
+		}
+		cloudInstanceID, vgID, err := splitID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+		vgC := st.NewIBMPIVolumeGroupClient(context.Background(), sess, cloudInstanceID)
+		vg, err := vgC.Get(vgID)
+		if err == nil {
+			log.Println("volume-group*****", vg.Status)
+			return fmt.Errorf("PI Volume Group still exists: %s", rs.Primary.ID)
+		}
+	}
+
+	return nil
+}
+func testAccCheckIBMPIVolumeGroupExists(n string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+
+		rs, ok := s.RootModule().Resources[n]
+
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return errors.New("No Record ID is set")
+		}
+
+		sess, err := acc.TestAccProvider.Meta().(conns.ClientSession).IBMPISession()
+		if err != nil {
+			return err
+		}
+		cloudInstanceID, vgID, err := splitID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+		client := st.NewIBMPIVolumeGroupClient(context.Background(), sess, cloudInstanceID)
+
+		_, err = client.Get(vgID)
+		if err != nil {
+			return err
+		}
+		return nil
+
+	}
+}
+
+func testAccCheckIBMPIVolumeGroupConfig(name string) string {
+	return fmt.Sprintf(`
+	resource "ibm_pi_volume_group" "power_volume_group"{
+		pi_volume_group_name       = "%[1]s"
+		pi_cloud_instance_id 	   = "%[2]s"
+		pi_volume_ids              = [""]
```

Comment:
Should this be allowed? I think vol ids are required field.

Reply:
yes, its a required filed. I removed the volume ids because of security issues, will add some dummy id.

Reply:
I hope we are not expecting dev to add volume ids before running the test. Don't add dummy values. Instead add a  ibm_pi_volume resource and use that id.

Reply:
changes done